tests/db_functions/text/test_regexpsubstr.py

```diff
@@ -0,0 +1,143 @@
+        queryset = Author.objects.annotate(altered_name=expression)
+        self.assertQuerysetEqual(queryset, [
+            ('George R. R. Martin', 'R. R. '),
+            ('J. R. R. Tolkien', self.empty),
+        ], transform=lambda x: (x.name, x.altered_name), ordered=False)
+
+    def test_lookbehind(self):
+        expression = RegexpSubstr('name', Value(r'(?<=George )(R\. ){2}'))
+        queryset = Author.objects.annotate(altered_name=expression)
+        self.assertQuerysetEqual(queryset, [
+            ('George R. R. Martin', 'R. R. '),
+            ('J. R. R. Tolkien', self.empty),
+        ], transform=lambda x: (x.name, x.altered_name), ordered=False)
+
+    def test_substitution(self):
+        expression = RegexpSubstr('name', Value(r'(R\. )\1'))
+        queryset = Author.objects.annotate(flipped_name=expression)
+        self.assertQuerysetEqual(queryset, [
+            ('George R. R. Martin', 'R. R. '),
+            ('J. R. R. Tolkien', 'R. R. '),
+        ], transform=lambda x: (x.name, x.flipped_name), ordered=False)
+
+    def test_expression(self):
+        expression = RegexpSubstr(Concat(Value('Author: '), 'name'), Value(r'(R\. ){2}'))
+        queryset = Author.objects.annotate(same_name=expression)
+        self.assertQuerysetEqual(queryset, [
+            ('George R. R. Martin', 'R. R. '),
+            ('J. R. R. Tolkien', 'R. R. '),
+        ], transform=lambda x: (x.name, x.same_name), ordered=False)
+
+    def test_update(self):
+        Author.objects.update(name=RegexpSubstr('name', Value(r'(Martin|Tolkien)')))
+        self.assertQuerysetEqual(Author.objects.all(), [
+            'Martin',
+            'Tolkien',
+        ], transform=lambda x: x.name, ordered=False)
+
+
+class RegexpSubstrFlagTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        Article.objects.create(
+            title='Chapter One',
+            text='First Line.\nSecond Line.\nThird Line.',
+            written=Now(),
+        )
+
+    @property
+    def empty(self):
+        mariadb = connection.vendor == 'mysql' and connection.mysql_is_mariadb
+        return '' if mariadb or connection.features.interprets_empty_strings_as_nulls else None
+
+    def test_dotall_flag(self):
+        expression = RegexpSubstr('text', Value(r'^.*$'), Value('s'))
+        article = Article.objects.annotate(result=expression).first()
+        self.assertEqual(article.result, 'First Line.\nSecond Line.\nThird Line.')
+
+    def test_multiline_flag(self):
+        expression = RegexpSubstr('text', Value(r'^.*\Z'), Value('m'))
+        article = Article.objects.annotate(result=expression).first()
+        self.assertEqual(article.result, 'Third Line.')
+
+    def test_extended_flag(self):
+        if connection.vendor == 'oracle':
+            # Oracle doesn't support comments in extended regular expressions.
+            pattern = Value(r"""
+                ^[^ ]*
+                \ Line\.
+            """)
```

Comment:
If it's valuable to test this case we should do this for all database and move a case with comments to a separate method (skipped on Oracle).